### DISCUSSION TITLE
Content based on subscription

### DISCUSSION THREAD

**Discussion by andriusmv** [supabase/supabase#6735] [Questions] [answered]
👋 newbie begginerDev here:

Question: How to restrict specific pages/components of my app based on Suscription and/or SubscriptionStatus? (Stripe Suscriptions for example).

I am following Vercel's template on NextJS-Subscriptions-Payments with Supabase (https://github.com/vercel/nextjs-subscription-payments), but I am clueless on how to restrict content once someone buys a specific Subscription.

I have a reusable component called AuthCheck.tsx to achieve this for logged-in users in general (email/password/GitHub/etc):

```
import Link from 'next/link';
import { useUser } from '@/utils/useUser';

const AuthCheck = (props: { children: any; fallback: any; }) => {
    const { user } = useUser();

    return user ? props.children : props.fallback || <Link href="/signin"><button>YOU MUST BE LOGGED IN TO SEEN THIS CONTENT</button></Link>;
}
    
    export default AuthCheck;
```
so my guess is that I should create a similar .tsx component for every Suscription I might have, for example Gold Member Suscription... GoldMembers.tsx that should look something similar:

```
import { Subscription } from 'types';
import Link from 'next/link';
import { useUser } from '@/utils/useUser';

const GoldMembers = (props: { children: any; fallback: any; }) => {
    const { Subscription } = Subscription();

    return Subscription ? props.children : props.fallback || <Link href="/signin"><button>YOU MUST SUBSCRIBE TO GOLD FIRST TO SEE THIS CONTENT</button></Link>;
}
    
    export default GoldMembers;
```

I have a types.ts file that contains this info:

```
import Stripe from 'stripe';
export interface PageMeta {
  title: string;
  description: string;
  cardImage: string;
}

export interface Customer {
  id: string /* primary key */;
  stripe_customer_id?: string;
}

export interface Product {
  id: string /* primary key */;
  active?: boolean;
  name?: string;
  description?: string;
  image?: string;
  metadata?: Stripe.Metadata;
}

export interface ProductWithPrice extends Product {
  prices?: Price[];
}

export interface UserDetails {
  id: string /* primary key */;
  first_name: string;
  last_name: string;
  full_name?: string;
  avatar_url?: string;
  billing_address?: Stripe.Address;
  payment_method?: Stripe.PaymentMethod[Stripe.PaymentMethod.Type];
}

export interface Price {
  id: string /* primary key */;
  product_id?: string /* foreign key to products.id */;
  active?: boolean;
  description?: string;
  unit_amount?: number;
  currency?: string;
  type?: string;
  interval?: Stripe.Price.Recurring.Interval;
  interval_count?: number;
  trial_period_days?: number | null;
  metadata?: Stripe.Metadata;
  products?: Product;
}

export interface PriceWithProduct extends Price {}

export interface Subscription {
  id: string /* primary key */;
  user_id: string;
  status?: Stripe.Subscription.Status;
  metadata?: Stripe.Metadata;
  price_id?: string /* foreign key to prices.id */;
  quantity?: number;
  cancel_at_period_end?: boolean;
  created: string;
  current_period_start: string;
  current_period_end: string;
  ended_at?: string;
  cancel_at?: string;
  canceled_at?: string;
  trial_start?: string;
  trial_end?: string;
  prices?: Price;
}
```

Any idea on this? Thanks a lot!

AM

**Comment by dijonmusters** [comment, marked as answer]
Hey Andrés, excellent question!

I think the way I would tackle this depends on how different each of the different tiers is. If they are just seeing slightly different content on the page, then I would use a single page and fetch the different content via `getServerSideProps`. If the content was significantly different then I would create a different page for each level of subscription, and enforce that the user is on that tier or redirect them in `getServerSideProps`.

The important part is you need to know who the user is on the server-side, which requires an auth cookie to be set anytime the user signs in. You can then check this cookie in `getServerSideProps` to ensure your user has the right permissions level.

I recommend checking out [my free egghead course](https://egghead.io/courses/build-a-saas-product-with-next-js-supabase-and-stripe-61f2bc20) for an example on how to implement this:

[This lesson](https://egghead.io/lessons/supabase-implement-gated-content-using-row-level-security-with-supabase) in particular should be able to nudge you in the right direction 👍
